### PR TITLE
Fix WebSocket connection bug

### PR DESCRIPTION
**Pull request by olexii4**
### What does this PR do?
Add default header before the open WebSocket connection.

### Which issue does this PR related to?
https://github.com/eclipse/che/issues/18477

Signed-off-by: Oleksii Orel <oorel@redhat.com>

### CODE REVIEW

**Comment by Ohrimenko1988** [comment]
[crw-ci-test]

**Comment by Ohrimenko1988** [comment]
[crw-ci-test]

**Comment by Ohrimenko1988** [comment]
[crw-ci-test]

**Comment by Ohrimenko1988** [comment]
[crw-ci-test]

**Comment by Ohrimenko1988** [comment]
[crw-ci-test]

**Comment by Ohrimenko1988** [comment]
[crw-ci-test]

**Comment by Ohrimenko1988** [comment]
[crw-ci-test]

**Comment by Ohrimenko1988** [comment]
[crw-ci-test]

**Comment by Ohrimenko1988** [comment]
[crw-ci-test]

**Comment by Ohrimenko1988** [comment]
[crw-ci-test]

**Comment by Ohrimenko1988** [comment]
[crw-ci-test]

File: src/services/workspace-client/CheWorkspaceClient.ts [hunk 1]
```diff
@@ -47,16 +49,16 @@ export class CheWorkspaceClient {
...
-      axios.interceptors.request.use(async request => {
+      this.axios.interceptors.request.use(async request => {
         const keycloak = KeycloakSetup.keycloakAuth.keycloak as any;
         if (keycloak && keycloak.updateToken && !isUpdated) {
           updateTimer();
           try {
             await new Promise((resolve, reject) => {
               keycloak.updateToken(5).success((refreshed: boolean) => {
-                if (refreshed && keycloak.token && axios.defaults && axios.defaults.headers && axios.defaults.headers.common) {
+                if (refreshed && keycloak.token && this.axios.defaults && this.axios.defaults.headers && this.axios.defaults.headers.common) {
```
Comment (akurinnoy) [inline src/services/workspace-client/CheWorkspaceClient.ts]: I'd use optional chaining here and set authorization header if it's provided in any case:
```suggestion
                if (refreshed && keycloak.token) {
                  if(this.axios.defaults?.headers?.common === undefined) {
                    this.axios.defaults = { headers: { common: {} };
                  }
```

File: src/services/workspace-client/CheWorkspaceClient.ts [hunk 2]
```diff
@@ -121,11 +123,15 @@ export class CheWorkspaceClient {
 
   async updateJsonRpcMasterApi(): Promise<void> {
     let jsonRpcApiLocation = this.originLocation.replace('http', 'ws') + this.websocketContext;
-    this._jsonRpcMasterApi = WorkspaceClient.getJsonRpcApi(jsonRpcApiLocation);
     // connect
     if (this.token) {
+      if (this.axios.defaults && this.axios.defaults.headers && this.axios.defaults.headers.common) {
```
Comment (akurinnoy) [inline src/services/workspace-client/CheWorkspaceClient.ts]: same as [above](https://github.com/che-incubator/che-dashboard-next/pull/86/files#r532467795)

**Comment by Ohrimenko1988** [comment]
[crw-ci-test]

File: src/services/workspace-client/CheWorkspaceClient.ts (hunk 1 above)
Comment (olexii4) [reply src/services/workspace-client/CheWorkspaceClient.ts]: I have fixed

File: src/services/workspace-client/CheWorkspaceClient.ts (hunk 2 above)
Comment (olexii4) [reply src/services/workspace-client/CheWorkspaceClient.ts]: I have fixed

**Review by akurinnoy — APPROVED** [review]

**Pull request merged**